packages/auth/lib/hashSessionId.js

```diff
@@ -0,0 +1,9 @@
+module.exports = async (id, salt) => {
+  const crypto = require('crypto')
+  return new Promise((resolve, reject) => {
+    crypto.pbkdf2(id, salt, 1000, 64, 'sha512', (err, derivedKey) => {
```

Comment:
- make it more expensive to calculate the hash by using pbkdf2
- don’t use a server side secret to generate the hash , as a bad guy could bruteforce the secret out of his own hash by comparing to the raw session id
- use the user’s email as salt to mitigate rainbow table attacks

@patte @tpreusse

Reply:
I'm still the opinion that it's so much more easy to trash this completely. Get rid of all the complications like async, like the recommended key length for pbkdf2 is 128bit (my email is shorter already) and just add a column to the sessions table called `id uuid not null`. This id is just used for referencing the session via the API, and public. It has nothing to do with the session id, that is just db.sid <-> cookie.